Samples/src/OracleSynthesis/OracleSynthesis.qs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.Samples.OracleSynthesis {
+
+    open Microsoft.Quantum.Primitive;
+    open Microsoft.Quantum.Canon;
+
+    /// # Summary
+    /// Computes Hadamard transform of a Boolean function in {-1,1} encoding
+    /// using Yates's method
```

Comment:
If at all possible, please provide a link to a publication or webpage (ideally with DOI) about this algorithm. 
You might want to use a reference tag, e.g., like this: 
\# References
 \- Title, authors, link to paper, etc

Reply:
Yes, I will add this.

Reply:
There is no DOI though